src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssKeysAndValues = sourceString
+    .split(';')
+    .filter((item) => item.trim().length)
+    .reduce((prev, item) => {
+      const index = item.indexOf(':');
+
+      return {
+        ...prev,
+        [item.slice(0, index).trim()]: item.slice(index + 1).trim(),
```

Comment:
Avoid using the spread operator inside the reduce callback to copy the accumulator object on each iteration. This can lead to performance issues due to the repeated creation of new objects. Consider using the `Object.assign` method or directly assigning new properties to the `prev` object. [CODE KNOWLEDGE]